nixos/modules/services/networking/zerobin.nix

```diff
@@ -0,0 +1,102 @@
+{ config, pkgs, lib, nodes, ... }:
+with lib;
+let
+  cfg = config.services.zerobin;
+
+  zerobin_config = pkgs.writeText "zerobin-config.py" ''
+  PASTE_FILES_ROOT = "${cfg.dataDir}"
+  ${cfg.extraConfig}
+  '';
+
+in
+  {
+    options = {
+      services.zerobin = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+          Enable the 0bin encrypted pastebin
+          '';
+        };
+
+        dataDir = mkOption {
+          type = types.string;
+          default = "/var/lib/zerobin";
+          description = ''
+          Path to the 0bin data directory
+          '';
+        };
+
+        user = mkOption {
+          type = types.string;
+          default = "zerobin";
+          description = ''
+          The user 0bin should run as
+          '';
+        };
+
+        group = mkOption {
+          type = types.string;
+          default = "zerobin";
+          description = ''
+          The group 0bin should run as
+          '';
+        };
+
+        listenPort = mkOption {
+          type = types.int;
+          default = 8000;
+          example = 1357;
+          description = ''
+          The port on which to listen
+          '';
+        };
+
+        listenAddress = mkOption {
+          type = types.string;
+          default = "localhost";
+          example = "127.0.0.1";
+          description = ''
+          The adress zerobin should listen to
```

Comment:
Typo: `/adress/address/`. Also, I think it's a bit confusing that you mention the daemon's name here but not in port, it gives the impression that they're referring to different entities. I suggest harmonizing the two.